201-front-door-session-affinity/azuredeploy.json

```diff
@@ -23,21 +23,15 @@
             "metadata": {
                 "description": "Whether to allow session affinity on this host. Valid options are 'Enabled' or 'Disabled'."
             }
-        },
-        "sessionAffinityTtlSeconds": {
-            "type": "int",
-            "metadata": {
-                "description": "The TTL to use in seconds for session affinity, if applicable."
-            }
-        }
+        }        
     },
     "variables": {
-        "frontdoorref": "[resourceId('Microsoft.Network/frontDoors', parameters('frontDoorName'))]",
-        "frontdoorLocation": "global"
+        "frontdoorLocation": "global",
+        "frontdoorSuffix": ".azurefd.net"
```

Comment:
use reference()

Reply:
The resource cannot make a reference to itself while is being created.